plugins/chat/lib/chat/notification_consolidation_extension.rb

```diff
@@ -0,0 +1,68 @@
+# frozen_string_literal: true
+
+module Chat
+  module NotificationConsolidationExtension
+    def self.chat_message_plan
+      Notifications::ConsolidateNotifications.new(
+        from: Notification.types[:chat_message],
+        to: Notification.types[:chat_message],
+        threshold: 1,
+        consolidation_window: 60.minutes,
+        unconsolidated_query_blk:
+          Proc.new do |notifications, data|
+            notifications.where("data::json ->> 'consolidated' IS NULL").where(
+              "data::json ->> 'chat_channel_id' = ?",
+              data[:chat_channel_id].to_s,
+            )
+          end,
+        consolidated_query_blk:
+          Proc.new do |notifications, data|
+            notifications.where("(data::json ->> 'consolidated')::bool").where(
+              "data::json ->> 'chat_channel_id' = ?",
+              data[:chat_channel_id].to_s,
+            )
+          end,
+      ).set_mutations(
+        set_data_blk:
+          Proc.new do |notification|
+            data = notification.data_hash
+
+            last_chat_message_notification =
+              Notification
+                .where(user_id: notification.user_id)
+                .order("notifications.id DESC")
+                .where(
+                  "data::json ->> 'chat_channel_id' = ?",
+                  notification.data_hash[:chat_channel_id].to_s,
+                )
+                .where(notification_type: Notification.types[:chat_message])
+                .where("created_at > ?", 1.hour.ago)
```

Comment:
maybe we put this in a constant with a clear name to document a little bit what is this one hour